### PR TITLE
config: chromeos: enable wifi basic test on Chromebooks

### PR DESCRIPTION
**Pull request by laura-nao**
Add basic test to verify basic WiFi functionality. Enable the test on ARM64 and x86_64 Chromebooks.

Requires: https://github.com/kernelci/kernelci-core/pull/2581

### CODE REVIEW

**Review by pawiecz — APPROVED** [review]
LGTM

(`nfsroot` used in the job description was verified to include the required dependencies)

**Comment by pawiecz** [comment]
(Most conflicts could be resolved by rebasing on https://github.com/kernelci/kernelci-pipeline/commit/396c67d0ba5260a56b4c4ceccd1635665150f527)

**Comment by JenySadadia** [comment]
Hello @laura-nao 
Please verify the details I've added here https://github.com/kernelci/kcidb/pull/538

**Comment by laura-nao** [comment]
> (Most conflicts could be resolved by rebasing on [396c67d](https://github.com/kernelci/kernelci-pipeline/commit/396c67d0ba5260a56b4c4ceccd1635665150f527))

Thanks for the heads up, I didn't notice there were conflicts. Should be fixed now!

**Comment by pawiecz** [comment]
> I guess I can remove the skip label and get it tested?

Label removed, should be picked up at the next staging run.

> I was waiting for approval on the jinja2 changes, to avoid breaking staging

PR should not cause conflicts but my merge commit should be dropped from it before pulling (maintainers beware!).
Let's get some WiFi tests results from staging instance (just as a confirmation) and then get rid of that merge commit.

**Comment by laura-nao** [comment]
Didn't find any result on staging yet (using the result_summary script). I'll try again tomorrow and report back.

**Comment by laura-nao** [comment]
I still don't see any result on staging (quickly checked on LAVA directly as well) - @pawiecz any idea on what may prevent these tests from being scheduled?

**Comment by pawiecz** [comment]
Unfortunately nothing yet, @JenySadadia and I [are investigating](https://kernelci.slack.com/archives/C015N3UF7C1/p1720698272477229).

**Comment by nuclearcat** [comment]
Removed

**Review by a-wai — APPROVED** [review]
LGTM

**Comment by pawiecz** [comment]
It looks like there might still be some references to deleted job templates:
```sh
$ grep -nrE "template: (kselftest.jinja2|sleep.jinja2|v4l2-decoder-conformance.jinja2|watchdog-reset.jinja2)"
pipeline.yaml:926:    template: kselftest.jinja2
pipeline.yaml:950:    template: kselftest.jinja2
pipeline.yaml:959:    template: kselftest.jinja2
jobs-chromeos.yaml:476:    template: kselftest.jinja2
```

**Comment by pawiecz** [comment]
I did not manage to complete pending reviews in time and this PR needed another rebase. It had been done as the https://github.com/kernelci/kernelci-pipeline/tree/wip/wifi-basic-test-support-fix

Thanks to merging #729 the only remaining [change to apply was in `config/jobs-chromeos.yaml`](https://github.com/kernelci/kernelci-pipeline/commit/51ba73c247e05f870cea2ca915a436245d3c8b07). Please consider squashing it with your initial refactor before pushing the updated branch to this PR.

**Comment by laura-nao** [comment]
@pawiecz done, thanks for the heads up! While I was at it, I also added a commit to update the references to the kselftest job definitions in the documentation.

File: config/jobs-chromeos.yaml [hunk 1]
```diff
@@ -376,13 +378,26 @@ _anchors:
...
     kind: job
     params: &watchdog-reset-job-params
+      test_method: watchdog-reset
       bl_message: 'coreboot-'
       wdt_dev: 'watchdog0'
     kcidb_test_suite: kernelci_watchdog_reset
 
+  wifi-basic: &wifi-basic-job
+    template: generic.jinja2
+    kind: test
```
Comment (a-wai) [inline config/jobs-chromeos.yaml:391]: Shouldn't that be `job`?

File: config/jobs-chromeos.yaml [hunk 2]
```diff
@@ -456,9 +473,11 @@ jobs:
     kcidb_test_suite: kselftest.acpi
 
   kselftest-device-error-logs:
-    template: kselftest.jinja2
+    template: generic.jinja2
     kind: test
```
Comment (a-wai) [inline config/jobs-chromeos.yaml:477]:
```suggestion
    kind: job
```

**Review by a-wai — COMMENTED** [review]
A minor nit but looks good overall, removing staging-skip so it can be tested.

File: config/jobs-chromeos.yaml (hunk 1 above)
Comment (laura-nao) [reply config/jobs-chromeos.yaml]: @a-wai indeed, I missed this. Fixed now!

**Review by a-wai — APPROVED** [review]

File: config/jobs-chromeos.yaml [hunk 3]
```diff
@@ -376,13 +378,26 @@ _anchors:
...
+      test_method: watchdog-reset
       bl_message: 'coreboot-'
       wdt_dev: 'watchdog0'
     kcidb_test_suite: kernelci_watchdog_reset
 
+  wifi-basic: &wifi-basic-job
+    template: generic.jinja2
+    kind: job
+    params: &wifi-basic-job-params
+      test_method: wifi-basic
```
[the remark this answers is not in the document]
Comment (laura-nao) [reply config/jobs-chromeos.yaml:393]: Right, fixed. I think it was there at some point but got lost in one of the numerous rebases :sweat: thanks!

**Review by pawiecz — APPROVED** [review]
LGTM

Verified on staging:
- https://staging.kernelci.org:9000/viewer?search=parent%3D66b93a563178e5bd67de18cb (ARM)
- https://staging.kernelci.org:9000/viewer?search=parent%3D66b93a523178e5bd67de18c8 (x86)

**Pull request merged**